keymap.c

```diff
@@ -1298,6 +1298,186 @@ enum CommandResult mutt_parse_bind(struct Buffer *buf, struct Buffer *s,
   return rc;
 }
 
+/**
+ * parse_menu - Parse a menu-names into array
+ * @param menu     Array for results
+ * @param s        string containing menu-names string
+ * @param maxmenus Total number of menus
+ * @param nummenus Number of menus this config applies to
+ * @param err      Buffer for error messages
+ * @param bind     If true 'bind', otherwise 'macro'
+ *
+ * Expects to see: <menu-string>[,<menu-string>]
+ */
+static void *parse_menu(int *menu, char *s, int maxmenus, int *nummenus,
+                        struct Buffer *err, char *cmd)
+{
+  int i = 0;
+  char *q = NULL;
+
+  /* menu name */
+  while (i < maxmenus)
+  {
+    q = strchr(s, ',');
+    if (q)
+      *q = '\0';
+
+    menu[i] = mutt_map_get_value(s, Menus);
+    if (menu[i] == -1)
+      mutt_buffer_printf(err, _("%s: no such menu"), s);
+    i++;
+    if (q)
+      s = q + 1;
+  }
+  *nummenus = i;
+  return NULL;
+}
+
+/**
+ * km_unbind_all - free all keys in Keymap from arg
+ * @param Keymap
+ * @param data
+ *
+ * Function iterate through Keymap and frees keys
+ * defined either by macro or bind
+ */
+static void km_unbind_all(struct Keymap **map, unsigned long mode)
+{
+  struct Keymap *cur, *next, *first, *last;
+
+  first = last = NULL;
+  cur = *map;
+
+  while (cur != NULL)
+  {
+    next = cur->next;
+    if ((mode & MUTT_UNBIND && cur->macro == NULL) ||
+        (mode & MUTT_UNMACRO && cur->macro != NULL))
```

Comment:
Wrap the individual conditions in `()` so we end up with `(mode & MUTT_UNBIND) && (cur->macro == NULL)...`